kinematics_interface/include/kinematics_interface/kinematics_interface_base.hpp

```diff
@@ -39,46 +41,49 @@ class KinematicsBaseClass
   /**
      * \brief Convert Cartesian delta-x to joint delta-theta, using the Jacobian.
      * \param[in] joint_pos joint positions of the robot in radians
-     * \param[in] delta_x_vec input Cartesian deltas (x, y, z, wx, wy, wz)
-     * \param[out] delta_theta_vec output vector with joint states
+     * \param[in] delta_x input Cartesian deltas (x, y, z, wx, wy, wz)
+     * \param[in] link_name the link name at which delta_x is applied
+     * \param[out] delta_theta outputs joint deltas
      * \return true if successful
      */
   virtual bool convert_cartesian_deltas_to_joint_deltas(
-    const std::vector<double> & joint_pos, const std::vector<double> & delta_x_vec,
-    const std::string & link_name, std::vector<double> & delta_theta_vec) = 0;
+    const Eigen::VectorXd & joint_pos, const Eigen::Matrix<double, 6, 1> & delta_x,
+    const std::string & link_name, Eigen::VectorXd & delta_theta) = 0;
 
   /**
      * \brief Convert joint delta-theta to Cartesian delta-x.
-     * \param joint_pos joint positions of the robot in radiansgit checkout
-     * \param[in] delta_theta_vec vector with joint states
-     * \param[out] delta_x_vec  Cartesian deltas (x, y, z, wx, wy, wz)
+     * \param joint_pos joint positions of the robot in radians
+     * \param[in] delta_theta joint deltas
+     * \param[in] link_name the link name at which delta_x is calculated
+     * \param[out] delta_x  Cartesian deltas (x, y, z, wx, wy, wz)
      * \return true if successful
      */
   virtual bool convert_joint_deltas_to_cartesian_deltas(
-    const std::vector<double> & joint_pos, const std::vector<double> & delta_theta_vec,
-    const std::string & link_name, std::vector<double> & delta_x_vec) = 0;
+    const Eigen::VectorXd & joint_pos, const Eigen::VectorXd & delta_theta,
+    const std::string & link_name, Eigen::Matrix<double, 6, 1> & delta_x) = 0;
+
 
   /**
     * \brief Calculates the joint transform for a specified link using provided joint positions.
     * \param[in] joint_pos joint positions of the robot in radians
     * \param[in] link_name the name of the link to find the transform for
-    * \param[out] transform_vec transformation matrix of the specified link in column major format.
+    * \param[out] transform transformation matrix of the specified link
     * \return true if successful
     */
   virtual bool calculate_link_transform(
-    const std::vector<double> & joint_pos, const std::string & link_name,
-    std::vector<double> & transform_vec) = 0;
+    const Eigen::VectorXd & joint_pos, const std::string & link_name,
+    Eigen::Matrix<double, 4, 4> & transform) = 0;
```

Comment:
This type can be used for transforms.
```suggestion
    Eigen::Isometry3d & transform) = 0;
```

Reply:
Done